packages/hmr-plugin/src/symbols.ts

```diff
@@ -1,22 +1,72 @@
 import { StateContext } from '@ngxs/store';
+import { NgModuleRef } from '@angular/core';
 
-export interface NgxsStoreSnapshot {
+import { HmrManager } from './hmr-manager';
+
+export interface NgxsHmrSnapshot {
   [key: string]: any;
 }
 
 export const NGXS_HMR_SNAPSHOT_KEY = '__NGXS_HMR_SNAPSHOT__';
 
-export interface NgxsHmrLifeCycle<T = NgxsStoreSnapshot> {
+export interface NgxsHmrLifeCycle<T = NgxsHmrSnapshot> {
   /**
    * hmrNgxsStoreOnInit is called when the AppModule on init
-   * @param context - StateContext for the current app state from Store
+   * @param ctx - StateContext for the current app state from Store
    * @param snapshot - previous state from Store after last hmr on destroy
    */
-  hmrNgxsStoreOnInit(context: StateContext<T>, snapshot: Partial<T>): void;
+  hmrNgxsStoreOnInit(ctx: StateContext<T>, snapshot: Partial<T>): void;
 
   /**
    * hmrNgxsStoreOnInit is called when the AppModule on destroy
-   * @param context - current state from Store
+   * @param ctx - StateContext for the current app state from Store
+   */
+  hmrNgxsStoreBeforeOnDestroy(ctx: StateContext<T>): Partial<T>;
+}
+
+export type CallStackFrame<T> = (ctx: StateContext<T>, prevState: Partial<T>) => void;
+export type BootstrapModuleType<T> = () => Promise<NgModuleRef<T>>;
+export type HmrAfterOnInit<T extends NgxsHmrLifeCycle<S>, S = NgxsHmrSnapshot> = (
+  manager: HmrManager<T, S>
+) => void;
+
+export interface NgxsHmrOptions<T extends NgxsHmrLifeCycle<S>, S = NgxsHmrSnapshot> {
+  /**
+   * @description
+   * clear log after each hmr update
+   * (default: true)
+   */
+  autoClearLogs?: boolean;
+
+  /**
+   * @description
+   * deferred time before loading the old state
+   * (default: 100ms)
    */
-  hmrNgxsStoreBeforeOnDestroy(context: StateContext<T>): T;
+  deferTime?: number;
+
+  /**
+   * @description - method to call intermediate result
+   * @param manager
+   */
+  hmrAfterOnInit?: HmrAfterOnInit<T, S>;
```

Comment:
Why is this necessary to expose `HmrAfterOnInit` as part of the options?
This would cause the manager and much of the internals to become public.

Reply:
`HmrAfterOnInit` - Because you want to control the process as a user.

That is, you want to understand when your method was executed to perform for example something else.

```ts
await hmr(module, bootstrap, { 
  hmrAfterOnInit: ({ store }) => {
    store.dispatch(new UPDATE_STATE_HMR()); // custom trigger all states if we want
  }
});
```

Reply:
Could we then only expose a `stateContext` as a parameter to this method?
Not any internal HMR plugin objects.

Reply:
then I don't know how to test it in my tests
https://github.com/ngxs/store/blob/fe4b138c1989ea667a3c1906f8d71c9c7624e1f2/packages/hmr-plugin/src/tests/hmr-manager.spec.ts

Reply:
You shouldn't be testing internals but rather verifying the state of the system through public methods.

Reply:
done